internal/testutils/nettest/free_port.go

```diff
@@ -0,0 +1,38 @@
+package nettest
+
+import "net"
+
+func getClosedTCPAddr() (*net.TCPAddr, error) {
+	address, err := net.ResolveTCPAddr("tcp", "127.0.0.1:0")
+	if err != nil {
+		return nil, err
+	}
+
+	listener, err := net.ListenTCP("tcp", address)
+	if err != nil {
+		return nil, err
+	}
+	defer listener.Close()
+
+	return listener.Addr().(*net.TCPAddr), nil
+}
+
+// MustGetFreeHostPort returns a TCP host:port that is free for unit tests
+// that cannot use port 0.
+func MustGetFreeHostPort() string {
```

Comment:
Note previous work:

https://github.com/yarpc/yarpc-go/pull/866
https://github.com/yarpc/yarpc-go/pull/897

Prashant and I have been talking offline, I'd like to fix this problem, 866 has a partial solution to fix the race too I think.